[dotnet] Add migration about serialization (#3490)
getsentry/sentry-docs: src/platforms/dotnet/migration.mdx

options.StackTraceMode = StackTraceMode.Original;
```

### Serialization

On version 3.0.0 the Sentry SDK for .NET changed from `Newtonsoft.Json` to `System.Text.Json`.

> This introduces some advantages, such as no dependency is needed (when targeting .NET Core 3.0, .NET 5 and later version) and much faster performance.

Author: I was hoping to describe that the SDK has "no dependency". Let me see if I can reword it:

> This introduces some advantages, such as no dependency is needed (when targeting .NET Core 3.0, .NET 5 and later version) and much faster performance.

Author: ```suggestion
Using the new library has many advantages. One of which is the lack of an additional dependency (when targeting .NET Core 3, .NET 5 and later versions) and much faster performance.
```

> This introduces some advantages, such as no dependency is needed (when targeting .NET Core 3.0, .NET 5 and later version) and much faster performance.

Author: Is the new version a bit better?

> One breaking change in behavior that came up was that `Newtonsoft.Json` was able to serialize some types that are not possible with `System.Text.Json`.
> This affects objects set as [Context](../enriching-events/context/) and `Extra`.

Reviewer: ```suggestion
One breaking change that affects objects set as [Context](../enriching-events/context/) and `Extra` is that `Newtonsoft.Json` was previously able to serialize some types, which is not possible with `System.Text.Json`.
```

### Serialization

In version 3.0.0, the Sentry SDK for .NET changed from `Newtonsoft.Json` to `System.Text.Json`.

> Using the new library has many advantages. One of which is the lack of an additional dependency (when targeting .NET Core 3, .NET 5 and later versions) and much faster performance.

Reviewer: ```suggestion
Using the new library has many advantages. These include the lack of an additional dependency (when targeting .NET Core 3, .NET 5 and later versions) and much faster performance.
```
